File: lib_recyclerviewadapter/src/main/java/lib/kalu/recyclerview/animation/SlideInBottomAnimation.java
package lib.kalu.recyclerview.animation;

import android.animation.Animator;
import android.animation.ObjectAnimator;
import android.view.View;

public final class SlideInBottomAnimation implements BaseAnimation {

    @Override
    public Animator[] getAnimators(View view) {

        final ObjectAnimator animator = ObjectAnimator.ofFloat(view, "translationY", view.getMeasuredHeight(), 0);
        return new Animator[]{animator};
    }
}

File: lib_kotlin/src/main/java/lib/kalu/adapter/animation/SlideInBottomAnimation.kt
package lib.kalu.adapter.animation

import android.animation.Animator
import android.animation.ObjectAnimator
import android.view.View

class SlideInBottomAnimation : BaseAnimation {

    override fun getAnimators(view: View): Array<Animator> {
        return arrayOf(ObjectAnimator.ofFloat(view, "translationY", view.measuredHeight.toFloat(), 0f))
    }
}
